### ISSUE TITLE
Terms of Use dialogue box

### ISSUE THREAD

**Issue by LlGC-szw** [UniversalViewer/user-stories#3]
## So that I know whether or not I can use this for the purpose intended

**Comment by LlGC-szw** [comment]
If using a URL the dialogue box isn't large enough to contain this. Also the user isn't compelled to read the terms before downloading.

**Comment by LlGC-szw** [comment]
![download_box_mockup_02](https://user-images.githubusercontent.com/10922825/32094065-7ec9f2ac-baf6-11e7-976d-b7a8730e52ac.png)

**Comment by tomcrane** [comment]
I think this is clickthrough auth on the PDF resource. In the same way that clickthrough auth works on the regular resources in many of Wellcome's manifests:

https://wellcomelibrary.org/item/b18311623

This way, you MUST have a cookie to access the PDF, and you acquire the cookie by interacting with an auth service. This would require you to implement a clickthrough auth service on your server (not that much work, just for clickthrough) but more significantly it requires that the UV implement auth on things that are not image services (discussed in Toronto WG) which is also a requirement for Stanford. Worth talking this through on the call.

**Comment by tomcrane** [comment]
Agreed on Community call that the Auth pattern is the way to go in the long term, but it requires an as yet unwritten addition to the auth spec and would also require NLW to implement an auth service.

So in the short term, the task is to reproduce the functionality visible in this example from the BL:

http://access.bl.uk/item/viewer/ark:/81055/vdc_00000004216E

![image](https://user-images.githubusercontent.com/1443575/32615025-7fc9e822-c566-11e7-8823-e666fe74e29c.png)

when you click "Terms of use" you get:

![image](https://user-images.githubusercontent.com/1443575/32615064-9d06d800-c566-11e7-9f67-bef9055fd77b.png)

...some UI tweaks required:

NLW want a link, as in @LlGC-szw example above
The dialogue box needs to be big enough to accommodate the link

**Comment by edsilv** [comment]
Ok to elide the link instead of making the box bigger? The boxes are deliberately small so they fit in small embeds. Eliding is usually the best strategy as links can be of arbitrary length.

**Comment by edsilv** [comment]
Hi @LlGC-szw 

Could you check this please? http://universalviewer.io/examples/#?c=0&m=0&s=0&cv=0&manifest=https%3A%2F%2Fedsilv.github.io%2Ftest-manifests%2Flonglicense.json&xywh=-1289%2C-106%2C5080%2C3750

**Comment by LlGC-szw** [comment]
Happy with this so far. We have noticed that the pop-up only show the English so we'll report that but we would like to test with our more wordy field in both languages before signing off on this.

**Comment by LlGC-szw** [comment]
Just to update you, we have staff off sick at the moment so no access to a manifest for testing.

**Comment by LlGC-szw** [comment]
The demo links are:

http://universalviewer.io/examples/#?c=0&m=0&s=0&cv=0&manifest=http%3A%2F%2Fwww.llgc.org.uk%2Frhannu%2FTestScif2176035.json&xywh=-1447%2C-94%2C5049%2C3726

and

http://universalviewer.io/examples/#?c=0&m=0&s=0&cv=0&manifest=http%3A%2F%2Fwww.llgc.org.uk%2Frhannu%2FTestCcymod1319153.json&xywh=-1622%2C-250%2C6765%2C4993

**Comment by LlGC-szw** [comment]
The link is showing (from 'license') in the pop-up but the link isn't clickable.

http://universalviewer.io/examples/#?c=0&m=0&s=0&cv=0&manifest=http%3A%2F%2Fwww.llgc.org.uk%2Frhannu%2FTestScif2176035.json&xywh=-511%2C-95%2C3177%2C3726

**Comment by LlGC-szw** [comment]
Looking at this example the terms dialogue is defaulting to attribution instead of the license field:

http://universalviewer.io/examples/#?c=0&m=0&s=0&cv=37&manifest=http%3A%2F%2Fwww.llgc.org.uk%2Frhannu%2FTestCcymod1319153.json&xywh=-2183%2C-2376%2C7882%2C9244

**Comment by ioanrichards** [comment]
Hi @edsilv, I've been told by @LlGC-szw that we need to move the license information into the root of the manifest, do you have a example to help please?

**Comment by tomcrane** [comment]
Hi @ioanrichards - the `license` field is a direct property of the manifest (if it applies to the whole manifest), and it must be a URI.

Here's a BL example:

http://api.bl.uk/metadata/iiif/ark:/81055/vdc_100022545251.0x000002/manifest.json

**Comment by ioanrichards** [comment]
Hi @tomcrane - thanks for the example, easy to see what is required.  One problem we have is not all license information is shown through a URI, some discussion required.

**Comment by edsilv** [comment]
If you have attribution and/or license, show the 'terms of use' link. In the dialogue, show attribution and license titles if present.

**Comment by LlGC-szw** [comment]
Just to let you know we have had a pdf generation issue so we won't be going live tomorrow with this :'( At least it gives us more time to get v3 up and running...

**Comment by LlGC-szw** [comment]
Could the remaining changes be done as open collective work?

**Comment by LlGC-szw** [comment]
This is what I had in mind (under the current layout).

![terms-of-use-box-mockup](https://user-images.githubusercontent.com/10922825/36204569-a68d458a-1183-11e8-9d72-185e32437372.png)

**Comment by LlGC-szw** [comment]
It has also been discussed that the link could be represented by an image in certain situations eg copyright statements.

**Comment by LlGC-szw** [comment]
Following our meeting this morning I'm going to tweak the user story slightly to reflect our needs more: 

As a National Library of Wales Journals user
I can find out what I can and can't do with the content I want to download
So that I know whether or not I can use this for the purpose intended

Acceptance criteria:
The user is informed of what they can and cannot do with the download

We're going to look at (as discussed in the community call) mapping the 'license' field to text and 'more info' links etc.

**Comment by hackartisan** [comment]
UniversalViewer/user-stories#3

Estimate: 1 hour
Added by tomcrane
